source/game/kart/KartSuspensionPhysics.cc

```diff
@@ -0,0 +1,235 @@
+        }
+    }
+
+    m_hitboxGroup->hitbox(0).setLastPos(center);
+
+    m_topmostPos = topmostPos;
+    m_wheelEdgePos = m_pos + m_effectiveRadius * move()->totalScale() * bottom;
+    m_effectiveRadius += (m_targetEffectiveRadius - m_effectiveRadius) * 0.1f;
+    m_suspTravel = bottom.dot(m_pos - topmostPos);
+}
+
+EGG::Vector3f WheelPhysics::pos() const {
+    return m_pos;
+}
+
+EGG::Vector3f WheelPhysics::lastPosDiff() const {
+    return m_lastPosDiff;
+}
+
+f32 WheelPhysics::suspTravel() {
+    return m_suspTravel;
+}
+
+EGG::Vector3f WheelPhysics::topmostPos() const {
+    return m_topmostPos;
+}
+
+CollisionGroup *WheelPhysics::hitboxGroup() {
+    return m_hitboxGroup;
+}
+
+const CollisionGroup *WheelPhysics::hitboxGroup() const {
+    return m_hitboxGroup;
+}
+
+EGG::Vector3f WheelPhysics::speed() const {
+    return m_speed;
+}
+
+f32 WheelPhysics::effectiveRadius() const {
+    return m_effectiveRadius;
+}
+
+void WheelPhysics::setSuspTravel(f32 suspTravel) {
+    m_suspTravel = suspTravel;
+}
+
+void WheelPhysics::setPos(const EGG::Vector3f &pos) {
+    m_pos = pos;
+}
+
+void WheelPhysics::setLastPos(const EGG::Vector3f &pos) {
+    m_lastPos = pos;
+}
+
+void WheelPhysics::setLastPosDiff(const EGG::Vector3f &pos) {
+    m_lastPosDiff = pos;
+}
+
+void WheelPhysics::setWheelEdgePos(const EGG::Vector3f &pos) {
+    m_wheelEdgePos = pos;
+}
+
+void WheelPhysics::set_48(const EGG::Vector3f &vec) {
+    _48 = vec;
+}
+
+KartSuspensionPhysics::KartSuspensionPhysics(u16 wheelIdx, u16 bspWheelIdx)
+    : m_tirePhysics(nullptr), m_bspWheelIdx(bspWheelIdx), m_wheelIdx(wheelIdx) {}
+
+KartSuspensionPhysics::~KartSuspensionPhysics() = default;
+
+void KartSuspensionPhysics::init() {
+    m_tirePhysics = tire(m_wheelIdx)->wheelPhysics();
+    m_bspWheel = &bsp().wheels[m_bspWheelIdx];
+}
+
+void KartSuspensionPhysics::reset() {
+    m_topmostPos.setZero();
+    m_maxTravelScaled = 0.0f;
+    m_bottomDir.setZero();
+}
+
+void KartSuspensionPhysics::setInitialState() {
+    EGG::Vector3f rotatedRelPos = dynamics()->fullRot().rotateVector(m_bspWheel->relPosition);
+    rotatedRelPos += pos();
+
+    EGG::Vector3f unitRotated = dynamics()->fullRot().rotateVector(-EGG::Vector3f::ey);
+
+    m_tirePhysics->setPos(rotatedRelPos + m_bspWheel->maxTravel * unitRotated);
+    m_tirePhysics->setLastPos(rotatedRelPos + m_bspWheel->maxTravel * unitRotated);
+    m_tirePhysics->setLastPosDiff(m_tirePhysics->pos() - rotatedRelPos);
+    m_tirePhysics->setWheelEdgePos(m_tirePhysics->pos() +
+            (m_tirePhysics->effectiveRadius() * move()->totalScale() * unitRotated));
+    m_tirePhysics->hitboxGroup()->hitbox(0).setWorldPos(m_tirePhysics->pos());
+    m_tirePhysics->hitboxGroup()->hitbox(0).setLastPos(pos() + 100 * EGG::Vector3f::ey);
+    m_topmostPos = rotatedRelPos;
+}
+
+void KartSuspensionPhysics::calcCollision(f32 dt, const EGG::Vector3f &gravity,
+        const EGG::Matrix34f &mat) { // mat right on 10th call
```

Comment:
Might be a few lingering comments. I think I got them, so will fix in next force-push.